File: code_samples/scala/utilities_date_time.scala
/**
 * Utilities: Date Time
 * AI/ML Training Sample
 */

class DateTime {
  private var data: String = ""
  
  def process(input: String): Unit = {
    data = input
  }
  
  def getData: String = data
  
  def validate: Boolean = data.nonEmpty
}

object DateTimeExample {
  def main(args: Array[String]): Unit = {
    val instance = new DateTime()
    instance.process("example")
    println(s"Data: ${instance.getData}")
    println(s"Valid: ${instance.validate}")
  }
}

File: code_samples/r/utilities_date_time.r
# Utilities: Date Time
# AI/ML Training Sample

DateTime <- setRefClass(
    "DateTime",
    fields = list(data = "character"),
    methods = list(
        initialize = function() {
            data <<- ""
        },
        process = function(input) {
            data <<- input
        },
        getData = function() {
            return(data)
        },
        validate = function() {
            return(nchar(data) > 0)
        }
    )
)

# Example usage
instance <- DateTime$new()
instance$process("example")
cat("Data:", instance$getData(), "\n")
cat("Valid:", instance$validate(), "\n")
